src/mpl/include/mpl_gpu.h

```diff
@@ -8,9 +8,12 @@
 
 #include "mplconfig.h"
 
+#undef MPL_HAVE_GPU
 #ifdef MPL_HAVE_CUDA
+#define MPL_HAVE_GPU MPL_GPU_TYPE_CUDA
 #include "mpl_gpu_cuda.h"
 #elif defined MPL_HAVE_ZE
+#define MPL_HAVE_GPU MPL_GPU_TYPE_ZE
 #include "mpl_gpu_ze.h"
 #else
 #include "mpl_gpu_fallback.h"
```

Comment:
This is a good idea. What do you think about putting the `static inline MPL_gpu_query_pointer_attr` function definition in `mpl_gpu_fallback.h`?

Reply:
Oh, looks like we'd miss the definition of `MPL_pointer_attr_t`. Oh well. I think this is good to go.

Reply:
> Oh, looks like we'd miss the definition of MPL_pointer_attr_t. Oh well. I think this is good to go.

Yep, exactly that :smile: